go_contacts/backends/contacts.py

```diff
@@ -91,6 +94,33 @@ def all_keys(self):
         """
         raise NotImplementedError()
 
+    def _reverse_delivery_class(self):
+        res = defaultdict(list)
+        for cls, value in DELIVERY_CLASSES.iteritems():
+            res[value['field']].append(cls)
+        return res
+
+    @inlineCallbacks
+    def _get_contacts_by_query(self, query):
+        try:
+            [field, value] = query.split('=')
+        except ValueError:
+            raise CollectionUsageError(
+                "Query must be of the form 'field=value'")
+        valid_keys = self.delivery_classes.keys()
+        if field not in valid_keys:
```

Comment:
And just use ``self.delivery_classes.keys()`` in the error message (perhaps sorted?).